10_1_graph.js

```diff
@@ -0,0 +1,15 @@
+export function buildGraph(edges) {
+    let graph = Object.create(null);
+    function addEdge(from, to) {
+        if (graph[from] == null) {
+            graph[from] = [to];
+        } else {
+            graph[from].push(to);
+        }
+    }
+    for (let [from, to] of edges) {
```

Comment:
using `map` is better than `for let`, this way you avoid possibility of mutating the array.
edges.map(edge => {
addEdge(graph, from, to)
});